### DISCUSSION TITLE
PM_fit$run error

### DISCUSSION THREAD

**Discussion by BoburaII** [LAPKB/Pmetrics#126] [Q&A] [answered]
Hi,
I am trying to run the PM_fit$run function with the model below;

mod2 <- PM_model$new(list(
  pri = list(
    Ka = range(0.00, 10.00),
    V = range(0.00, 100.00),
    Kcp = range (0.00, 5.00),
    Kpc = range(0.00, 5.00),
    Ke = range(0.00, 5.00)
  ),
  out = list(
    Y1 = list(
      value = "X(2)/V",
      error = list(
        model = proportional(5.000),
        assay = c(0.100, 0.100, 0.000, 0.000)
      )
    )
  )
))

I get this error:

Error: Symbol 'ka' at (1) already has basic type of REAL genmodel.for:242:18:
242
REAL*8 Ка, Ка, V,V, Кср, Кср, Крс, Крс, Ке, Ке
1
Error: Symbol 'ka' at (1) already has basic type of REAL
Error:
You have Fortran syntax errors in your model statments, as detailed above.
In addition: Warning message:
In system(syntaxcheck, intern = T) :
running command 'gfortran -fsyntax-only genmodel.for' had status 1

Models that did run fine previously, now return the same error.

Any help with this? 
Thanks.

**Comment by mnneely** [comment, marked as answer]
To avoid confusion with the built in “range” function, we changed the function to be ab().

So your model declaration would look like this:

mod2 <- PM_model$new(list(
pri = list(
Ka = ab(0.00, 10.00),
...

If you are getting more errors, install the current version with:

devtools::install_github("LAPKB/Pmetrics").



Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Bonn90 ***@***.***>
Sent: Thursday, December 1, 2022 8:39:41 AM
To: LAPKB/Pmetrics ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [LAPKB/Pmetrics] PM_fit$run error (Discussion #126)


Hi,
I am trying to run the PM_fit$run function with the model below;

mod2 <- PM_model$new(list(
pri = list(
Ka = range(0.00, 10.00),
V = range(0.00, 100.00),
Kcp = range (0.00, 5.00),
Kpc = range(0.00, 5.00),
Ke = range(0.00, 5.00)
),
out = list(
Y1 = list(
value = "X(2)/V",
error = list(
model = proportional(5.000),
assay = c(0.100, 0.100, 0.000, 0.000)
)
)
)
))

I get this error:

Error: Symbol 'ka' at (1) already has basic type of REAL genmodel.for:242:18:
242
REAL*8 Ка, Ка, V,V, Кср, Кср, Крс, Крс, Ке, Ке
1
Error: Symbol 'ka' at (1) already has basic type of REAL
Error:
You have Fortran syntax errors in your model statments, as detailed above.
In addition: Warning message:
In system(syntaxcheck, intern = T) :
running command 'gfortran -fsyntax-only genmodel.for' had status 1

Models that did run fine previously, now return the same error.

Any help with this?
Thanks.

**Reply by BoburaII** [reply]
Thank you! ab() sorts it out.
